File: Ruby/GameState.rb
#encoding:utf-8

####################################### 
# Álvaro Maldonado Medina             #      
# 2D- D3                              #
# Fichero: Irrgarten/GameState.rb     #
#######################################


module Irrgarten

    require_relative 'Player.rb'
    require_relative 'Labyrinth.rb'
    require_relative 'Monster.rb'
    require_relative 'Game.rb'

    class GameState

        #Constructor

        def initialize(labyrinth, players, monsters, current_player, winner,
                       log)
            
            @labyrinth = labyrinth.to_s
            @players = players.to_s
            @monsters = monsters.to_s
            @current_player = current_player.to_s
            @winner = winner.to_s
            @log = log.to_s


        end

        # Consultores

        def get_labyrinth
            @labyrinth.to_s
        end

        def get_players
            @players
        end

        def get_monsters
            @monsters
        end

        def get_current_player
            @current_player.to_s
        end

        def get_winner
            @winner.to_s
        end

        def get_log
            @log.to_s
        end

    end

end

File: Java/src/irrgarten/GameState.java
///////////////////////////////////////
// Álvaro Maldonado Medina     	     //
// 2D - D3                           //
// Fichero: irrgarten/GameState.java //
///////////////////////////////////////

package irrgarten;


public class GameState {
    
    private String labyrinth;
    private String players;
    private String monsters;
    private int currentPlayer;
    private boolean winner;
    private String log;
    
    
    // Constructor
    
    public GameState(String labyrinth, String players, String monsters, 
                     int currentPlayer, boolean winner, String log){
        
        this.labyrinth = labyrinth;
        this.players = players;
        this.monsters = monsters;
        this.currentPlayer = currentPlayer;
        this.winner = winner;
        this.log = log;
    }
    
    // Métodos get y set de la clase GameState
    
    public String getLabyrinth(){
        return labyrinth;
    }
    
    public String getPlayers(){
        return players;
    }
    
    public String getMonsters(){
        return monsters;
    }
    
    public int getCurrentPlayer(){
        return currentPlayer;
    }
    
    public boolean getWinner(){
        return winner;
    }
    
    public String getLog(){
        return log;
    }
}
